be/src/column/array_view_column.h

```diff
@@ -0,0 +1,229 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+
+#include "column/array_column.h"
+#include "column/column.h"
+#include "column/fixed_length_column.h"
+#include "column/nullable_column.h"
+#include "column/vectorized_fwd.h"
+
+namespace starrocks {
+
+class ArrayViewColumn final : public ColumnFactory<Column, ArrayViewColumn> {
```

Comment:
Good. The main differences between `ArrayViewColumn` and `ArrayColumn` is `replicate` method not to duplicate `_elements` but rather duplicate `offsets` and `length` to decrease data replications, am I right?

Reply:
> Good. The main differences between `ArrayViewColumn` and `ArrayColumn` is `replicate` method not to duplicate `_elements` but rather duplicate `offsets` and `length` to decrease data replications, am I right?

yeah, not only the `replicate` method, but also `append` and `assign` methords, they are similar. We want to avoid modifying elements in `ArrayViewColumn`